core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -392,7 +402,8 @@ class AuthorizerIntegrationTest extends BaseRequestTest {
       ApiKeys.DELETE_ACLS -> deleteAclsRequest,
       ApiKeys.DESCRIBE_ACLS -> describeAclsRequest,
       ApiKeys.ALTER_REPLICA_DIR -> alterReplicaDirRequest,
-      ApiKeys.DESCRIBE_LOG_DIRS -> describeLogDirsRequest
+      ApiKeys.DESCRIBE_LOG_DIRS -> describeLogDirsRequest,
+      ApiKeys.CREATE_PARTITIONS -> createPartitionsRequest
```

Comment:
@ijuma this is a really useful test, but it disturbs me that it was so easily forgotten. I wonder – given that many of the `*Response`s return per-topic errors – if we should add an assertion to this test:

    val notPerTopicErrorResponse = Set(
      ApiKeys.SASL_HANDSHAKE,
      ApiKeys.DELETE_RECORDS,
      ApiKeys.ADD_OFFSETS_TO_TXN,
      ApiKeys.INIT_PRODUCER_ID,
      ApiKeys.END_TXN,
      ApiKeys.ADD_PARTITIONS_TO_TXN,// actually this has per-topic errors
      ApiKeys.SASL_AUTHENTICATE,
      ApiKeys.TXN_OFFSET_COMMIT,// actually this has per-topic errors
      ApiKeys.LIST_GROUPS,
      ApiKeys.API_VERSIONS,
      ApiKeys.WRITE_TXN_MARKERS
    )
    val missing = ApiKeys.values.toSet -- notPerTopicErrorResponse -- requestKeyToRequest.keySet
    assertTrue(s"All ApiKeys whose response has per-topic errors should be tested, but ${missing.mkString(",")} aren't being tested", missing.isEmpty)

That way, when people add new `ApiKeys` the test fails and they at least have to think about whether to add their key to `notPerTopicErrorResponse` or the `requestKeyToRequest` map.

Reply:
Sounds great.